addons/binding/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/handler/OneWireGPIOHandler.java

```diff
@@ -0,0 +1,164 @@
+                        channelUID.getId());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            checkConfiguration();
+            updateStatus(ThingStatus.ONLINE);
+            startAutomaticRefresh();
+            super.initialize();
+
+        } catch (Exception e) {
+            logger.warn("Problem initializing thing: {}.", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    }
+
+    /**
+     * This method checks if the provided configuration is valid.
+     * When invalid parameter is found, default value is assigned.
+     */
+    private void checkConfiguration() {
+        logger.debug("Start reading Feed Thing configuration.");
+        Configuration configuration = getConfig();
+
+        // It is not necessary to check if the URL is valid, this will be done in fetchFeedData() method
+        gpioBusFile = (String) configuration.get(GPIO_BUS_FILE);
+        if (gpioBusFile == null) {
+            logger.warn("GPIO_BUS_FILE not set. Please check configuration, and set proper path to w1_slave file.");
+            throw new IllegalArgumentException("Invalid path to w1_slave file");
+        }
+
+        refreshTime = ((Number) configuration.get(REFRESH_TIME)).intValue();
+        if (refreshTime.intValue() <= 0) {
+            logger.warn("Refresh time [{}] is not valid. Falling back to default value: {}.", refreshTime,
+                    DEFAULT_REFRESH_TIME);
+            refreshTime = DEFAULT_REFRESH_TIME;
+        }
+    }
+
+    private void startAutomaticRefresh() {
+
+        Runnable refresher = new Runnable() {
+            @Override
+            public void run() {
+                List<Channel> channels = getThing().getChannels();
+                for (Channel channel : channels) {
+                    publishSensorValue(channel.getUID());
+                }
+            }
+        };
+
+        scheduler.scheduleWithFixedDelay(refresher, 0, refreshTime.intValue(), TimeUnit.SECONDS);
+        logger.debug("Start automatic refresh every {} seconds", refreshTime.intValue());
+    }
+
+    private void publishSensorValue(ChannelUID channelUID) {
+        String channelID = channelUID.getId();
+        State state = null;
+        switch (channelID) {
+            case TEMPERATURE:
+                BigDecimal temp = readSensorTemperature(gpioBusFile);
+                temp = temp.setScale(2, BigDecimal.ROUND_HALF_UP);
+                state = new DecimalType(temp);
+                break;
+        }
+        if (state != null) {
+            updateState(channelID, state);
+        } else {
+            logger.debug("Can not update channel with ID : {} - channel name might be wrong!", channelID);
+        }
+    }
+
+    BigDecimal readSensorTemperature(String gpioFile) {
+
+        try (Stream<String> stream = Files.lines(Paths.get(gpioFile))) {
```

Comment:
Paths#get might throw an InvalidPathException stating the path is invalid. This is not handled here and the Thing stays ONLINE all the time despite no value can be read.